src/main/java/io/vertx/ext/amqp/impl/AmqpConnectionImpl.java

```diff
@@ -265,20 +272,34 @@ public AmqpConnection receiver(String address, AmqpReceiverOptions receiverOptio
         receiver.setQoS(ProtonQoS.valueOf(receiverOptions.getQos().toUpperCase()));
       }
 
-      runWithTrampoline(x -> new AmqpReceiverImpl(address, this, receiver, handler, completionHandler));
+      runWithTrampoline(x -> {
+        new AmqpReceiverImpl(address, this, receiver, handler, completionHandler);
+      });
     });
     return this;
   }
 
   @Override
-  public AmqpConnection sender(String address, Handler<AsyncResult<AmqpSender>> completionHandler) {
+  public AmqpConnection createSender(String address, Handler<AsyncResult<AmqpSender>> completionHandler) {
+    Objects.requireNonNull(address, "The address must be set");
```

Comment:
I don't think I would require this. Is there a particular benefit?

Reply:
Ignore, you actually already changed it later to make it work with the other method, makes sense as it is now.